test/index.test.js

```diff
@@ -866,5 +866,28 @@ describe('mongoose module:', function() {
       const m = new mongoose.Mongoose();
       assert.deepEqual(await m.syncIndexes(), {});
     });
+    it('Allows a syncIndexes global option (gh-11030)', async function() {
+      const m = new mongoose.Mongoose();
+
+      m.set('debug', true);
+      const db = await m.connect('mongodb://localhost:27017/mongoose_test_11030');
+      const schema = new m.Schema({
+        title: { type: String, index: true }
+      });
+      const syncFalse = db.model('Sync', schema, 'Sync');
+      await syncFalse.init();
+      const Indexes = await syncFalse.listIndexes();
+      console.log(Indexes);
+      assert.equal(Indexes.length, 2);
+      await syncFalse.collection.createIndex({ name: 1 });
+      delete m.connection.models['Sync'];
+      m.set('syncIndexes', true);
+      const syncSchema = new m.Schema({ title: String });
```

Comment:
In order to test that `syncIndexes()` correctly adds new indexes as well as drops old indexes, let's add a property that isn't `name` or `title`, and add an index to that new property in the schema.